app/services/slack_service.rb

```diff
@@ -0,0 +1,183 @@
+module SlackService
+  class InvalidRequest < RuntimeError;
+  end
+  class InvalidCommand < RuntimeError;
+  end
+
+  public
+
+  def create_post(command_text, team_id, user_id)
+    slack_users = command_text.scan(/(?=<).*?(?<=>)/)
+
+    receivers = []
+
+    slack_users.each do |slack_user|
+      id = slack_user[slack_user.index('@') + 1..slack_user.index('|') - 1]
+      user = User.where(slack_id: id).take
+
+      if user == nil
+        name = slack_user[slack_user.index('|') + 1..slack_user.index('>') - 1]
+        raise InvalidCommand.new "#{name} has not connected their account to Slack."
+      end
+
+      receivers << user
+    end
+
+    if receivers.length == 0
+      raise InvalidCommand.new "Did you forget to mention any users with the '@' symbol?"
+    end
+
+    message = command_text[/'(.*?)'/m, 1]
+
+    if message == nil || message == ""
+      raise InvalidCommand.new 'Did you include a message surrounded by \'?'
+    end
+
+    amount = command_text.split(' ').last
+
+    if amount == nil || amount == ""
+      raise InvalidCommand.new 'Did you include an amount?'
+    end
+
+    begin
+      Integer(amount)
+    rescue ArgumentError
+      raise InvalidCommand.new 'Did you include an amount?'
+    end
+
+    team = Team.where(:slack_team_id => team_id).take
+
+    if team == nil
+      raise InvalidCommand.new 'This workspace does not have an associated Kudo-o-matic team, contact an admin'
+    end
+
+    sender = User.where(:slack_id => user_id).take
+
+    if sender == nil
+      raise InvalidCommand.new 'Your Slack account is not linked to Kudo-o-matic, use the /register command'
+    end
+
+    Post.new(
+        message: message,
+        amount: amount,
+        sender: sender,
+        receivers: receivers,
+        team: team,
+        kudos_meter: team.active_kudos_meter
+    )
+  end
```

Comment:
Whoa this is a long function, can you try breaking it apart into a few logical pieces?